dm/master/server_test.go

```diff
@@ -162,35 +162,26 @@ var (
 	etcdTestCli2          *clientv3.Client
 )
 
+type testMaster struct {
+	workerClients   map[string]workerrpc.Client
+	saveMaxRetryNum int
+	testT           *testing.T
+}
+
 func TestMaster(t *testing.T) {
 	err := log.InitLogger(&log.Config{})
 	if err != nil {
 		t.Fatal(err)
 	}
-
-	testEtcdCluster = integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1})
-	defer testEtcdCluster.Terminate(t)
-	etcdTestCli = testEtcdCluster.RandClient()
-
-	testEtcdCluster2 = integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1})
-	defer testEtcdCluster2.Terminate(t)
-	etcdTestCli2 = testEtcdCluster2.RandClient()
+	_ = check.Suite(&testMaster{testT: t})
```

Comment:
this is not a regular way to initialize a Suite. If you want to avoid concurrent running some cases of a Suite, you could using another struct instead of testMaster, and assign some test to the new struct.

Reply:
i am not  avoid concurrent running some cases, i want to use new `ectdclient` in eatch test case, but new `etcdclient` need `*testing.T` and `pingcap/checker` suit don't have this

Reply:
**BTW** we can use `check.Suite()` in  any where before `check.TestingT`
because the func `Suite()` just register our test struct to go check's  private variable  `allParallelSuites`

you can see more detail here https://github.com/go-check/check/blob/v1/run.go#L20

Reply:
https://github.com/pingcap/dm/blob/e120956ce9ef240a06e291518d40243fc303e739/pkg/etcdutil/etcdutil_test.go#L177

could initiate an etcd inside a test function.

https://github.com/pingcap/dm/blob/e120956ce9ef240a06e291518d40243fc303e739/dm/master/server_test.go#L171

or save them to many variables

Reply:
@lichunzhu @lance6716  i know we can do this manually in each test, but why? 

i used to use `testify` to write golang test , it also has the struct `suit` and provide friendly access to `*testing.T`

more info here:
https://github.com/stretchr/testify/blob/v1.7.0/suite/suite.go#L29

Reply:
> i know we can do this manually in each test

what's the meaning of "this"?

Reply:
I'm requesting `_ = check.Suite` should be called at program initiation time

Reply:
"this" mean `could initiate an etcd inside a test function.`
i think it is better to use suite's hook: `befortest/teardowntest` to init/clean `etcd client` 

i don't konw why  if  putting `check.Suite()` in `TestMaster` will cause `check.Suite()` be called mutiple times?
please tell me the reason

Reply:
moved `check.Suite()` out of `TestMust`